modules/phase_field/src/materials/MathFreeEnergy.C

```diff
@@ -0,0 +1,50 @@
+#include "MathFreeEnergy.h"
+
+template<>
+InputParameters validParams<MathFreeEnergy>()
+{
+  InputParameters params = validParams<DerivativeBaseMaterial>();
+  params.addClassDescription("Material that implements the math free energy and its derivatives: \nF = 1/4(1 + c)^2*(1 - c)^2");
+  params.addRequiredCoupledVar("c","Concentration variable");
+  return params;
+}
+
+MathFreeEnergy::MathFreeEnergy(const std::string & name,
+                       InputParameters parameters) :
+    DerivativeBaseMaterial(name, parameters),
+    _c(coupledValue("c")),
+    _c_var(coupled("c"))
+{}
+
+Real
+MathFreeEnergy::computeF()
+{
+  return 1.0/4.0*(1.0 + _c[_qp])*(1.0 + _c[_qp])*(1.0 - _c[_qp])*(1.0 - _c[_qp]);
+}
+
+Real
+MathFreeEnergy::computeDF(unsigned int j_var)
+{
+  if (j_var == _c_var)
```

Comment:
I don't think we need all the derivative materials to use the actual var numbers, just the interface that a user sees when they are coding in their own derivatives. Since DerivativeParsedMaterialHelper overrides computeProperties, it is only materials that inherit directly from DerivativeBaseMaterial that use the computeDF, computeD2F, etc. functions. So, if we add the _arg_number in the computeProperties from DerivativeBaseMaterial, everything that inherits from DerivativeParsedMaterialHelper can stay the way it is.